File: webservices/src/main/java/fieg/modulos/boleto/enums/SituacaoBoleto.java
package fieg.modulos.boleto.enums;

import fieg.core.interfaces.EnumBanco;


public enum SituacaoBoleto implements EnumBanco {

    PAGO_MAIOR("PAGO MAIOR"),
    PAGO_MENOR("PAGO MENOR"),
    PAGO("PAGO"),
    EM_ABERTO("EM ABERTO"),
    CANCELADO("CANCELADO"),
    PAGO_CADIN("PAGO CADIN"),
    ATIVO("ATIVO"),
    REMOVIDO("REMOVIDO");

    private final String valorBanco;

    SituacaoBoleto(String valorBanco) {
        this.valorBanco = valorBanco;
    }

    public static SituacaoBoleto fromString(String situacao) {
        for (SituacaoBoleto s : SituacaoBoleto.values()) {
            if (s.getValorBanco().equalsIgnoreCase(situacao)) {
                return s;
            }
        }
        throw new IllegalArgumentException("Situação inválida: " + situacao);
    }

    @Override
    public String getValorBanco() {
        return valorBanco;
    }
}

File: web/src/models/DTOs/ConsultaBoletoCaixa/SituacaoBoleto.ts
export interface SituacaoBoletoDTO {
  descricao: string;
  nomeEnum: string;
}

export interface SituacaoBoletoType {
  REMOVIDO: SituacaoBoletoDTO;
  EM_ABERTO: SituacaoBoletoDTO;
  ATIVO: SituacaoBoletoDTO;
  PAGO_MAIOR: SituacaoBoletoDTO;
  PAGO_MENOR: SituacaoBoletoDTO;
  PAGO: SituacaoBoletoDTO;
  CANCELADO: SituacaoBoletoDTO;
  PAGO_CADIN: SituacaoBoletoDTO;
}

export type SituacaoBoleto = keyof SituacaoBoletoType;

/**
 * Aqui, label é um nome amigável e value é o valor no Enum SituacaoBoleto no cr5-webservices-v2.
 * Constante feita para ser usada como opção de SelectControl
 */
export const situacoesBoletoValues: SituacaoBoletoType = {
    REMOVIDO: {
        nomeEnum: "REMOVIDO",
        descricao: "Removido",
    },
    EM_ABERTO: {
        nomeEnum: "EM_ABERTO",
        descricao: "Em Aberto",
    },
    ATIVO: {
        nomeEnum: "ATIVO",
        descricao: "Ativo",
    },
    PAGO_MAIOR: {
        nomeEnum: "PAGO_MAIOR",
        descricao: "Pago Maior",
    },
    PAGO_MENOR: {
        nomeEnum: "PAGO_MENOR",
        descricao: "Pago Menor",
    },
    PAGO: {
        nomeEnum: "PAGO",
        descricao: "Pago",  
    },
    CANCELADO: {
        nomeEnum: "CANCELADO",
        descricao: "Cancelado",
    },
    PAGO_CADIN: {
        nomeEnum: "PAGO_CADIN", 
        descricao: "Pago Cadin",
    },
};
